Backfill genome data

joverlee521 opened the pull request:
Adds scripts necessary to parse through previously generated genomic data that have not been uploaded to ID3C. 

The process would be as follows:
1. Parse sequence reads sets and upload them via `id3c sequence-read-set upload`.
2. Convert batch FASTA files to ndjson
3. Parse assembly summary data
4. Combine all three (sequence_read_set.ndjson, consensus_genome.ndjson, summary.ndjson)
5. Upload the final ndjson file via `id3c consensus-genome upload`.

tsibley commented on scripts/backfill_genome_data/sequence_read_sets.py:
```diff
@@ -0,0 +1,48 @@
+
+All sequence read sets are output to stdout as newline-delimited JSON
+records.  You will likely want to redirect stdout to a file.
+"""
```
How does this differ from `id3c sequence-read-set parse`?

joverlee521 replied:
I believe they are the same. I wasn't sure which side to add the parse step.

tsibley replied:
Nod. I think it's useful to have a way in ID3C to scan a set of fastqs and create sequence read sets for them. This is a likely a situation we'll have to deal with again and again.

tsibley commented on scripts/backfill_genome_data/sequence_read_sets.py:
```diff
@@ -0,0 +1,48 @@
+        description = __doc__,
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+    parser.add_argument("directory", type=str, nargs='?',
```
Why `nargs='?'`? I think that's only useful when there's also a `default` or `const` value.

Ditto for the other scripts in this PR.

joverlee521 committed:
Add script to parse sequence read sets

This is used to backfill sequence read sets within ID3C. 
The sequence read sets are printed out as 
newline-delimited JSON records. The ndjson file can then
be passed to the `id3c sequence-read-set upload` to be
uploaded into `warehouse.sequence_read_set`.

joverlee521 committed:
Add script to parse assembly summary data

Takes in bowtie2 and BAMstats directories to parse 
through all the files within them to create 
newline-delimited JSON records of summary data
for each sample/reference pair. 

This is to be used in the backfill genome data 
process.

joverlee521 committed:
Add script to convert FASTA to ndjson

Takes in a batch FASTA file and coverts each sequence
to a newline-delimited JSON record. 

This is to be used in the backfill genome data 
process.

joverlee521 committed:
Add script to combine summary and genomes

This will take all of the ndjson files created by sister scripts
to then create a final newline-delimited JSON record that
contains an entire consensus genome along with its 
summary data. This prints out ndjson records that match
the JSON records created by `upload_consensus_genomes_to_id3c`.

To be used in the backfill genome data process.

joverlee521 committed:
Skip "Undetermined" files in sequence read sets.

tsibley commented:
With the genomes branch deployed, this looks ready to go to me.

I do think that `scripts/backfill_genome_data/sequence_read_sets.py` is unnecessary given `id3c sequence-read-set parse`, and as a general rule I like to delete unnecessary code. But fine to leave it as well, esp. if it serves a purpose I'm missing.

@joverlee521 Do you want to be in charge of running the backfill, or should I pick that up?

joverlee521 commented:
I'm in the middle of doing the backfill on rhino. I've kept `sequeunce-read-set parse` just so it's easier for me to run on rhino directly.
